[
  {
    "project": "openvinotoolkit/datumaro",
    "commit": "f11071f3a109f674bdefea782543e49e305c529a",
    "filename": "datumaro/components/dataset.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/openvinotoolkit-datumaro/datumaro/components/dataset.py",
    "file_hunks_size": 3,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "datumaro/components/dataset.py:784:12 Unbound name [10]: Name `on_error` is used but not defined in the current scope.",
    "message": " Name `on_error` is used but not defined in the current scope.",
    "rule_id": "Unbound name [10]",
    "warning_line_no": 784,
    "warning_line": "            on_error.do(shutil.rmtree, save_dir, ignore_errors=True)"
  }
]